### PR TITLE
Add Symbol constructor to nodiff

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

**Review by willtebbutt — APPROVED** [review]
LGTM -- just needs a patch bump.

File: src/rulesets/Base/nondiff.jl [hunk 1]
```diff
@@ -374,6 +374,7 @@ VERSION >= v"1.1" && @non_differentiable splitpath(::AbstractString)
 @non_differentiable success(::Base.Process)
 @non_differentiable success(::Base.ProcessChain)
 @non_differentiable supertype(::Any)
+@non_differentiable Symbol(::Any)
```
Comment (mzgubic) [inline src/rulesets/Base/nondiff.jl:377]:
```suggestion
@non_differentiable Symbol(::Any...)
```

**Review by mzgubic — APPROVED** [review]

**Comment by oxinabox** [comment]
@ChrisRackauckas  merge and tag whenever you want

**Comment by DhairyaLGandhi** [comment]
Mind if I?

**Comment by ChrisRackauckas** [comment]
I'd mind.

**Comment by ChrisRackauckas** [comment]
😄

**Pull request merged**